web/src/main/java/org/akaza/openclinica/controller/StudyEventController.java

```diff
@@ -239,55 +256,89 @@ public RestfulServiceHelper getRestfulServiceHelper() {
     @ApiOperation(value = "To schedule an event for participants at study level in bulk",  notes = "Will read the information of SudyOID,ParticipantID, StudyEventOID, Ordinal, Start Date, End Date")
 	@ApiResponses(value = {
 	        @ApiResponse(code = 200, message = "Successful operation"),
 	        @ApiResponse(code = 400, message = "Bad Request -- Normally means Found validation errors, for detail please see the error list: <br /> ")})
 	@RequestMapping(value = "clinicaldata/studies/{studyOID}/events/bulk", method = RequestMethod.POST,consumes = {"multipart/form-data"})
+    @Async
 	public ResponseEntity<Object> scheduleBulkEventAtStudyLevel(HttpServletRequest request, 
 			MultipartFile file,
 			@PathVariable("studyOID") String studyOID) throws Exception {
 		
+    	 CompletableFuture<ResponseEntity<Object>> future = CompletableFuture.supplyAsync(() -> {
+    		 UserAccountBean ub = getUserAccount(request);    		 
+    		 return scheduleEvent(request,file, studyOID, null, ub);
+    	 });
+    	 
+    	 String uuid = UUID.randomUUID().toString();
+         
+         synchronized (expiringMap) {
+              expiringMap.put(uuid, future);
+          }
+        
+        ResponseEntity response = null;
+        RestReponseDTO responseDTO = new RestReponseDTO();
+  		String finalMsg = "The schedule job is running, here is the schedule job ID:" + uuid;
+  		responseDTO.setMessage(finalMsg);
+  		response = new ResponseEntity(responseDTO, org.springframework.http.HttpStatus.OK);
+  		
+  		return response;
 		
-    	return scheduleEvent(request,file, studyOID, null);
+    	
 	}
     
 
 
-	private ResponseEntity<Object> scheduleEvent(HttpServletRequest request,MultipartFile file, String studyOID, String siteOID) {
-		
+	private ResponseEntity<Object> scheduleEvent(HttpServletRequest request,MultipartFile file, String studyOID, String siteOID,UserAccountBean ub) {
+			
 		ResponseEntity response = null;
+		String logFileName = null;
 		
-		/**
-         * log error / info into log file 
-         * Response should be a logfile with columns - 
-         * Row, ParticipantID, StudyEventOID, Ordinal, Status, ErrorMessage
-         */ 
-    	String logFileName = null;
-    	if(siteOID == null) {
-    		logFileName = studyOID+"_log.txt";
-    	}else {
-    		logFileName = studyOID+"_"+siteOID+"_log.txt";
-    	}
+		if (!file.isEmpty()) {
+			 String fileNm = file.getOriginalFilename();
+			 
+			 //only support CSV file
+			 if(!(fileNm.endsWith(".csv")) ){
+				 throw new OpenClinicaSystemException("errorCode.notSupportedFileFormat","The file format is not supported at this time, please send CSV file, like *.csv ");
+			 }
+	
+			 int endIndex = fileNm.indexOf(".csv"); 
+			 String originalFileNm =  fileNm.substring(0, endIndex);
+			 logFileName = this.getRestfulServiceHelper().getMessageLogger().getLogfileNamewithTimeStamp(originalFileNm);
+			
+		}else {
+			throw new OpenClinicaSystemException("errorCode.emptyFile","The file is empty ");
```

Comment:
Please log the exception before throwing. We don't know what happens after it is thrown.